crates/proof-of-sql/src/base/database/column_operation.rs

```diff
@@ -398,6 +505,63 @@ where
     ))
 }
 
+/// Divide an owned column by another.
+///
+/// Notes:
+/// 1. We do not check for length equality here.
+/// 2. We use floor division for rounding.
+/// 3. If division by zero occurs, we return an error.
+/// 4. Precision and scale follow T-SQL rules. That is,
+///   - new_scale = max(6, right_precision + left_scale + 1)
+///   - new_precision = left_precision - left_scale + right_scale + new_scale
+pub(crate) fn try_divide_decimal_columns<S, T0, T1>(
+    lhs: &[T0],
+    rhs: &[T1],
+    left_column_type: ColumnType,
+    right_column_type: ColumnType,
+) -> ColumnOperationResult<(Precision, i8, Vec<S>)>
+where
+    S: Scalar + core::convert::From<T0> + core::convert::From<T1>,
+    T0: Copy + Debug,
+    T1: Copy + Debug,
+{
+    let len = lhs.len();
+    let new_column_type = try_divide_column_types(left_column_type, right_column_type)?;
+    let new_precision_value = new_column_type
+        .precision_value()
+        .expect("numeric columns have precision");
+    let new_scale = new_column_type.scale().expect("numeric columns have scale");
+    let lhs_scale = left_column_type
+        .scale()
+        .expect("numeric columns have scale");
+    let rhs_scale = right_column_type
+        .scale()
+        .expect("numeric columns have scale");
+    let res: Vec<S> = (0..len)
+        .map(|i| -> ColumnOperationResult<S> {
```

Comment:
NIT: Since `i` is only indexing the `lhs` and `rhs`, you could do this:
```suggestion
    let res: Vec<S> = lhs.iter().zip(rhs)
        .map(|(l, r)| -> ColumnOperationResult<S> {
```